ValveKeyValue/ValveKeyValue/ObjectCopier.cs

```diff
@@ -390,6 +384,12 @@ static object ConvertValue(object value, Type valueType, IObjectReflector reflec
 
         static bool TryConvertValueTo<TValue>(string name, object value, out TValue converted)
         {
+            if (typeof(TValue) == typeof(IntPtr))
+            {
+                converted = (TValue)(object)(IntPtr)(KVValue)value;
```

Comment:
![image](https://user-images.githubusercontent.com/613331/195932338-cd001803-8f84-4fbc-a737-bbe653a7436f.png)

Reply:
It looks ugly as hell in C# but it gets optimised by the runtime perfectly:

https://sharplab.io/#v2:C4LghgzgtgPgAgJgIwFgBQcAMACOSB0ASgK4B2wAllAKb4DCA9lAA4UA21ATgMpcBuFAMbUIAbnRZs3ABZhOzADJgARkTKUa4tBIDMuBNjroA3umznsAbQBSFYAHFqpLkIAUwAJ7NqDAGauKcgBKIIBdMwsbO0dnTjdPbz9XAElyAAVgThDwtAtcPQAVADUwNmJqbABZAB5i0vKAPlcAaSKSsoqADyDsU1y8iwpfbHcvH386jp6AXmnsBPGU9MyQiIHetfWLOAB2EcnyoNcGZQArakFgI9TgDKzOrS3sAF9NvLftvdcD6iDGUj4XGA9FkpAA5tQCmNXJ0ADTzMZJH4hR4WV5odG6fTYVrtcomTaWSrUYDSBgAE2SLDYrmJpIpVOYbAA8sxKAxSBB8AA5BipNiBQJgsKbOB6PAANmw1E6TKEdmwDG8nDAwAYnGwNzuLTa9QqfBmDXm0k4DAA7thSMQ2GxRNh0M8gA

```csharp
[JitGeneric(typeof(IntPtr))]
public TValue M<TValue>(KVValue x) {
    if (typeof(TValue) == typeof(IntPtr))
    {
        return (TValue)(object)(IntPtr)x;
    }

    return (TValue)Convert.ChangeType(x, typeof(TValue));
}
```

```text
C.M[[System.IntPtr, System.Private.CoreLib]](KVValue)
    L0000: mov ecx, edx
    L0002: call dword ptr [0x1ba2c98c] // this calls op_Explicit
    L0008: ret
```

Reply:
Good to know.